docs/articles/documentation/test-api/selecting-page-elements/selectors/creating-selectors.md

```diff
@@ -38,9 +38,8 @@ You can initialize a selector with any of these objects.
     const submitButton = Selector('#submit-button');
     ```
 
-* A regular function. Executed on the client side. Must return a DOM node, array of DOM nodes,
-  NodeList, HTMLCollection, `null` or `undefined`. Cannot use outer-scope variables from test code.
-  This is convenient when you need to use some client-side logic to get an element.
+* A regular function executed on the client side. This function must return a DOM node, an array of DOM nodes,
```

Comment:
For me it's not obvious here that I can pass some arguments to this function and in which cases I should you use `dependencies` instead of arguments

Reply:
I don't remember these reasons either 😄 
What are they?

Reply:
We should use dependencies for object which have to be serialized. [Here ](https://github.com/DevExpress/testcafe/pull/3666/files#diff-aed991e564624039815624cec861bf3bR35) you said correct e.g. `functions, variables or objects`

Reply:
It doesn't matter, all the arguments and dependencies get serialized:

https://github.com/DevExpress/testcafe/blob/master/src/client-functions/client-function-builder.js#L87